xchange-core/src/main/java/org/knowm/xchange/service/account/AccountService.java

```diff
@@ -125,4 +128,21 @@ default TradeHistoryParams createFundingHistoryParams() {
   default List<FundingRecord> getFundingHistory(TradeHistoryParams params) throws IOException {
     throw new NotYetImplementedForExchangeException();
   }
+  /**
+   * Get the trading fees per currency pair as determined by the given exchange's rules for
+   * adjusting fees by recent volume traded. Some exchanges will provide the current fees per
+   * currency via a single API request, while others require more logic to compute by hand.
+   *
+   * @return map between currency pairs and their fees at the time of invocation.
+   * @throws ExchangeException - Indication that the exchange reported some kind of error with the
+   *     request or response
+   * @throws NotAvailableFromExchangeException - Indication that the exchange does not support the
+   *     requested function or data
+   * @throws NotYetImplementedForExchangeException - Indication that the exchange supports the
+   *     requested function or data, but it has not yet been implemented
+   * @throws IOException - Indication that a networking error occurred while fetching JSON data
+   */
+  public default Map<CurrencyPair, Fee> getDynamicTradingFees() throws IOException {
+    throw new NotYetImplementedForExchangeException();
+  }
```

Comment:
how many exchanges provide this type of endpoint?
I don't think this is a generic feature for now

Reply:
Virtually all exchanges asses fees as a function of trailing 30 day volume. The brackets and the way that the trailing volume is computed vary by exchange, which is why the API exists. Some exchanges provide an explicit way of querying this information, while for others it is required to derive it from trailing 30 day volume and published fee brackets.

Reply:
I think this is a needed addition.

Reply:
ok